recursion/compiler/src/ir/builder.rs

```diff
@@ -49,18 +50,11 @@ impl<T> TracedVec<T> {
     /// Pushes a value to the vector and records a backtrace if SP1_DEBUG is enabled
     pub fn trace_push(&mut self, value: T) {
         self.vec.push(value);
-        match std::env::var("SP1_DEBUG")
-            .unwrap_or("false".to_string())
-            .to_lowercase()
-            .as_str()
-        {
-            "true" => {
-                self.traces.push(Some(Backtrace::new_unresolved()));
-            }
-            _ => {
-                self.traces.push(None);
-            }
-        };
+        if sp1_debug_mode() {
+            self.traces.push(Some(Backtrace::new_unresolved()));
+        } else {
+            self.traces.push(None);
```

Comment:
Is it necessary to have this vector of `None`?

Reply:
I think the old runtime may panic because it will try to look up a trace by index in `traces`.

Reply:
Oh I see now that this is the code that was here before. I think you can leave it as it for now.